huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go

```diff
@@ -0,0 +1,410 @@
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+		JSONBody:         utils.RemoveNil(buildCreateOrUpdateCatalogBodyParams(d)),
+	}
+
+	createResp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio catalog: %s", err)
+	}
+
+	createRespBody, err := utils.FlattenResponse(createResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	id, err := jmespath.Search("data.value.id", createRespBody)
+
+	if err != nil || id == nil {
+		return diag.Errorf("error creating DataArts Studio catalog: ID is not found in API response")
+	}
+
+	// need to set qualified ID to filter result in READ.
+	qualifiedID, err := jmespath.Search("data.value.qualified_id", createRespBody)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio catalog: qualifiedID is not found in API response")
+	}
+	d.SetId(id.(string))
+	d.Set("qualified_id", qualifiedID)
+
+	return resourceCatalogRead(ctx, d, meta)
+}
+
+func buildCreateOrUpdateCatalogBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"name":        d.Get("name"),
+		"owner":       d.Get("owner"),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"parent_id":   utils.ValueIngoreEmpty(d.Get("parent_id")),
+		"prev_id":     utils.ValueIngoreEmpty(d.Get("prev_id")),
+		"next_id":     utils.ValueIngoreEmpty(d.Get("next_id")),
+	}
+}
+
+func resourceCatalogRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/design/biz/catalogs/tree"
+		product = "dataarts"
+		mErr    *multierror.Error
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "DataArts Studio catalog")
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	paths := strings.Split(d.Get("qualified_id").(string), ".")
+	jsonPaths := fmt.Sprintf("data.value[?id=='%s']", paths[0])
+	for i, path := range paths {
+		if i == 0 {
+			continue
+		}
+		jsonPaths += fmt.Sprintf("[children][][?id=='%s'][]", path)
+	}
+
+	catalogs := utils.PathSearch(jsonPaths, getRespBody, make([]interface{}, 0)).([]interface{})
+	if len(catalogs) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "DataArts Studio catalog")
+	}
+
+	catalog := catalogs[0]
+	d.SetId(utils.PathSearch("id", catalog, "").(string))
```

Comment:
how about take this code into the import method?

Reply:
fixed.